File: Leetcode/238 - Product of Array Except Self/solution.py
def productExceptSelf(nums: List[int]) -> List[int]:
    res: List[int] = [1] * len(nums)
    prefix: int = 1
    suffix: int = 1
    
    i, j = 1, len(nums) - 2
    while i < len(nums) and j >= 0:
        prefix *= nums[i - 1]
        suffix *= nums[j + 1]
        res[i] *= prefix
        res[j] *= suffix
        i += 1
        j -= 1
    
    return res

File: Leetcode/238 - Product of Array Except Self/solution.ts
function productExceptSelf(nums: number[]): number[] {
    const res: number[] = new Array(nums.length).fill(1);
    let prefix: number = 1, suffix: number = 1;
    
    for (let i = 1, j = nums.length - 2; i < nums.length && j >= 0; i++, j--) {
        prefix *= nums[i - 1];
        suffix *= nums[j + 1];
        res[i] *= prefix;
        res[j] *= suffix;
    }
    
    return res;
};